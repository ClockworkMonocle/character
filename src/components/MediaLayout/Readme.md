Align items side-by-side, with a slight gap between them and vertical alignment configurable.

```js
require('../Type');

<MediaLayout
  key="test-1"
  media={<div style={{ height: '100px', width: '100px', backgroundColor: 'purple' }}/>}
  description={[
    <Type>What do you think about this purple?</Type>,
    <Type variant="subtext">this is some really awesome content</Type>
  ]}
/>
```
```js
require('../Type');

<MediaLayout
  key="test-2"
  align="top"
  media={<div style={{ height: '100px', width: '100px', backgroundColor: 'purple' }}/>}
  description={<Type>Align to the top</Type>}
/>
```

```js
require('../Type');

<MediaLayout
  key="test-3"
  align="bottom"
  media={<div style={{ height: '100px', width: '100px', backgroundColor: 'purple' }}/>}
  description={<Type>Align to the bottom</Type>}
/>
```
