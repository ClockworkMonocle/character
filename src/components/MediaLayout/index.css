
.media_layout--container {
  width: 100%;
  height: 100%;;
  display: flex;
  flex-direction: row;
}

.media_layout--top {
  align-items: flex-start;
}

.media_layout--middle {
  align-items: center;
}

.media_layout--bottom {
  align-items: flex-end;
}

.media_layout--media {
  margin-right: 8px;
}

.media_layout--desc {
  flex-grow: 1;
}
