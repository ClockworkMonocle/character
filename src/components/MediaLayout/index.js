import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import './index.css';

export default function MediaLayout({ media, description, align, className, ...props }) {
  const css = cn(
    'media_layout--container',
    `media_layout--${align}`,
    className
  );
  return (
    <div className={css}>
      <div className="media_layout--media">{media}</div>
      <div className="media_layout--desc">{description}</div>
    </div>
  );
}

MediaLayout.propTypes = {
  align: PropTypes.oneOf(['top', 'middle', 'bottom']),
  media: PropTypes.node,
  description: PropTypes.node,
};

MediaLayout.defaultProps = {
  align: 'middle',
};
