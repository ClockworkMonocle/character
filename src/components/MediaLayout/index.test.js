import React from 'react';
import { shallow } from 'enzyme';
import MediaLayout from './index';

test('<MediaLayout /> renders', () => {
  expect(shallow(<MediaLayout
    media={<div>media</div>}
    description={<div>description</div>}
  />)).toMatchSnapshot();
});
