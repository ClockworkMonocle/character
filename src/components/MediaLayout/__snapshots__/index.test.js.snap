// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediaLayout /> renders 1`] = `
<div
  className="media_layout--container media_layout--middle"
>
  <div
    className="media_layout--media"
  >
    <div>
      media
    </div>
  </div>
  <div
    className="media_layout--desc"
  >
    <div>
      description
    </div>
  </div>
</div>
`;
